### ISSUE TITLE
Access to "New" Quests. Scoring limiter or quest lines coherence?

### ISSUE THREAD

**Issue by AquariusPower** [Attnam/ivan#484]
Did Xinroch Tomb alternative "main" quest was imported from CLIVAN?

I was reading this https://attnam.com/wiki/Quests

the encrypted scroll, determines we wont be able to play or Main (GC) or Xinroch quest line. 
I mean "won't" because it means less time playing, less fun in a single play-thru.

Is such prevention just means to limit the game ending player's score?
I mean, just for the scoring we are unable to use a single character to play the whole/most content possible?
If so, if we could opt to not score, would we be able to access both quest lines (GC & XT) w/o breaking both quests concepts/coherence, are they depending on each other or are they essentially independent?

Also, the side quests, cool things we can do w/o ending the game.
I wonder how importable are the 5 quests from the variants? They seem to be all side quests right?
Essentially this mean more content on a single play-thru.
Unless there are more access restrictions only because of scoring? :(, or am I wrong?

EDIT: 
A good reason to chose one or other quest line is the player's power progression, in a sense that, weak mobs from upper dungeon's levels will have no chance against a player's character power that have already visited another dungeon deepest level (and survived).

What other games usually do is level up the mobs to make them not too easy to fight against, but I think Ivan does not do that or I am wrong?

So in short, I guess after the player acquire enough power (stats and items), the game gets too easy (? I never ended the game tho), and to avoid getting boring, a good (and cool) option is to end the game's quest line (the main ones) and go to the score board.

**Comment by fejoa** [comment]
The Tomb of Xinroch was developed on this version of IVAN, not in CLIVAN, but the map location was available in CLIVAN. 
The chances of the player succeeding scales, in part, with the number of items available in the game. In CLIVAN, which was derived from LIVAN, the number of items was higher, which contributed to making a playthrough of Gloomy Caves somewhat easier. I extended LIVAN by writing the new material in CLIVAN. While most of my work was shit, and I have no intention of porting the dungeons from there to here, it was worthwhile sandboxing and made me aware of the kind of game experience that players expect from IVAN.

At the moment, Gloomy Caves and Tomb of Xinroch are totally separate from one another in order to preserve the balance and aesthetic of GC as intended by the original devs, and to demonstrate the development of a different dungeon, hopefully with a different look and feel, but still IVAN.
I also purposefully kept Tomb of Xinroch separate from GC in case my work was shit again. It seems to have survived, but I wonder if most people play it, and I suspect it is being too similar to Gloomy Caves. @red-kangaroo has done a lot of great work to improve Tomb of Xinroch, and I really appreciate that.

To understand where the present day rationale has come from, you have to go back to [this thread started by SquashMonster](https://attnam.com/posts/20176). In particular, the roadmap outlined in [this post](https://attnam.com/posts/20179). This idea was _exceedingly_ well received by the community.

Based on SquashMonster's idea, and with the development done as catalogued in [this thread](https://attnam.com/topics/Where-to-next-with-worldmap-and-new-dungeon-locations), we are roughly at Phase 2. It was made possible to [add dungeon locations to the worldmap](https://attnam.com/wiki/Adding_Cities_and_Dungeons_to_the_IVAN_World_Map), without the need to compile the game, which was a response to demand from the community, where developers wanted to make modifications via script files only. There are a couple of tasks remaining, but I haven't found time to complete them. Nonetheless, it functions to a minimum spec.

The restriction between dungeons is there firstly to preserve Gloomy Caves, and secondly to enable development of dungeons and cities that are related to the IVAN world, and are just as immersive as what was originally authored. That the encrypted scroll is used for this at the moment is just a gum solution. One of the things IVAN needs is a generic facility for recording a dungeon's story state. I suppose that is as simple as introducing a new integer variable, so it's probably trivial. I think the Tomb of Xinroch has some story state stuff. Limiting the player's score was also intentional, because in CLIVAN, you had the opportunity to slay more creatures, and that is how most of the power players have managed to get such stupendously high highscores. I think highscores can be qualified by dungeon, in addition to score value. That way, varying scores from dungeons with different difficulty curves can be distinguished from one another.

The restrictions do not preclude another developer from making, for example, three new cities and dungeons and allowing them each to interact together, this is certainly possible. Indeed, it is desirable, and the original devs imagined a future IVAN featuring a political subsystem with interaction between cities. The reason we haven't got as far as that is because no-one has yet proposed a coherent method for doing this. Given the past experience of the forks (LIVAN, CLIVAN, IVANX, IVANT), it was found that the balance of the original game was very easily changed in subtle ways, and more often with the unintended consequence of making it easier. The only thing about several locations interacting is the risk of accidentally creating pedantic fetch-quest story mechanics, which was one of the criticisms of the nefarious "Ommel blood quest" in CLIVAN. I guess some fetch-quest is okay, but not if it is the only thing the player gets to do. 

My position is to preserve Gloomy Caves as best we can, and for developers to come along and strike out in a new direction, chiefly by composing novel, immersive dungeons and cities that attract players back to IVAN, increasing its fame and glory. I advocate for this way because it maximizes the opportunity for developers to build their own stuff without disturbing the original story line that fans are familiar with.

Of course, I don't get the final say, and some other devs may wish to add stuff. Having a git repo also means that changes can be made and things optimized, hopefully in a way that is guided by player experience. Certainly anything I have contributed may be re-written or deleted, and I wouldn't feel worried or sentimental about it, that's just what open-source means to me.

If you wanna know, I have a plan for a new dungeon, [Urok](https://github.com/fejoa/ivan/tree/Urok). I bags it. I did some stuff to navigate between open levels, but I haven't populated it with creatures, or designed the progression mechanics for it. I know roughly how I want it to work. I pledge it to be enjoyable, and it will be a magnum opus of sorts, but I doubt if I will make any more big dungeons after that.

**Comment by AquariusPower** [comment]
> the balance of the original game was very easily changed in subtle ways, and more often with the unintended consequence of making it easier

I like the crafting PR as much as it also scares me :)
The enhanced lights (far ones visible) is a natural expected thing, but makes things easier too.
May be, some features could block scoring? like in ex.: if any of these 2 are enabled, scoring is disabled, until both are better balanced. I don't really care for scoring, but I do for challenge, and making things easier scare me a lot, no bravery ring will help xD

> My position is to preserve Gloomy Caves as best we can, and for developers to come along and strike out in a new direction, chiefly by composing novel, immersive dungeons and cities that attract players back to IVAN, increasing its fame and glory. I advocate for this way because it maximizes the opportunity for developers to build their own stuff without disturbing the original story line that fans are familiar with.

Now I see how important it is to keep or GC or XT or anything else new, and the encrypted scroll is actually a great way to do that IMHO!

So the only way to have multiple quest lines in a single play would be to **grant dungeons of the next chosen quest line wont be easy**, and I think that is impossible, unless if **the player character could marry, grow old and unable to fight anymore** and share some just a bit of stuff with his children, like letting we chose only a single item from the inventory (not all items would be available tho), and give access to a new quest. 

That child would grow and be the new player, with the **same base/starting stats** we have on a new game nowadays, but that game would have a background story being the all previously played quest lines.
And such starting stats would grant the balance on new quest lines w/o having to go the weird path MMOs do of leveling NPCs to match current player power...

So back to the previous quest line: depending on how much you score on that, the player will have a certain amount of children, each being one chance to retry the current quest line in case the one dies on it's path.
That continuation would be from the quest line beginning and with a granted (?) bones file from the previous death, exclusive for that game line.

This way, the feeling of uncertainty and starting with almost nothing, could be brought back, and at the same time keep the progress of the previous quest line in the same savegame.

> Certainly anything I have contributed may be re-written or deleted, and I wouldn't feel worried or sentimental about it, that's just what open-source means to me.

Agreed! I expect that from whatever I implemented, let other ppl improve it. If the game gets better, it is worth any code changes :)

> I have a plan for a new dungeon, Urok. I bags it. I did some stuff to navigate between open levels

I was thinking on creating a way to let we "enter" any world map square, creating a single wilderness dungeon map, letting us hunt, may be build a farm one day with lots of more code, but... that would break the player's power progress before GC or XT and one day Urok :), and I couldnt think a way to make it more valid, appealing or balanced yet, so I didnt even start coding it. May be a small chance to find a single wild animal could work.

**Phase3 at https://attnam.com/posts/20179**
> Start adding higher-level-monster content (monsters) to the dungeons. Add higher-level variants of fixed monsters like Elpuri and the Enner Beast. Once there's enough we can do "ultra" endings for completing multiple plots in a row. Note that the game's **existing dynamic difficulty system** is important to making this work

the "existing dynamic difficulty system", mmm...
this may be what could allow multiple quest lines at once, also the "higher-level variants" would be a way to try to grant things dont get too easy, but it would still break the starting feeling when we have nothing and practically no skills.
I think the main reason I abandon so many gameplays is because the game starting feeling is the most cool part IMHO :)
After that, when we can carry a huge lot of items, I end spending a lot of time on the inventory because I have too many new options (and we also need them), and when that gets boring.. I just start another game xD
So the "procreation system" could be a good workaround between quest lines in the same gameplay keeping the starting feeling of uncertainty :)

**Comment by AquariusPower** [comment]
> The only thing about several locations interacting is the risk of accidentally creating pedantic fetch-quest story mechanics, which was one of the criticisms of the nefarious "Ommel blood quest" in CLIVAN. I guess some fetch-quest is okay, but not if it is the only thing the player gets to do.

I havent played CLIVAN to know but I guess...

a random event could create a specific quest necessity

that event would randomly affect the quest giver
- what was stolen? by who? can he be found (random tracking)? where?
- what was destroyed? what is required to rebuild? can we build it? 
- what difficulties arises from that destruction? can we help on overcoming them (sub-quests)?
etc

I think the more randomization, the more complex and hard to balance it gets
but I see no other way to prevent repetitiveness :)
